assets/js/product.js

```diff
@@ -129,15 +135,67 @@ storefrontApp.controller('productController', ['$rootScope', '$scope', '$window'
             });
         }
 
+        function partyComparePropertyMaps(propMap1, master) {
+            return _.every(_.keys(master), function (x) {
+                var retVal = propMap1.hasOwnProperty(x);
+                if (retVal) {
+                    retVal = propMap1[x][0].value == master[x][0].value;
+                }
+                return retVal;
+            });
+        }
+
+
+        function filterVariationBySelectedProps(variations, selectedPropMap) {
+            return _.filter(variations, function (x) {
+                return partyComparePropertyMaps(getVariationPropertyMap(x), selectedPropMap);
+            });
+        }
+
+        function isPropertyAvailable(property, key) {
+            var allVariationsMaps = angular.copy($scope.allVariationPropsMap);
+            _.each(allVariationsMaps[key], function (prop) {
+                prop.selected = false;
+            });
+            var workProperty = _.find(allVariationsMaps[key], function (y) { return y.value === property.value; });
+            workProperty.selected = true;
+            var variations = filterVariationBySelectedProps(allVariations, getSelectedPropsMap(allVariationsMaps));
+            if(variations.length) {
+                return true;
+
+            }
+            return false;
+            
+        }
+
         //Method called from View when user clicks one property value
         $scope.checkProperty = function (property) {
             //Select appropriate property and unselect previous selection
+            
+           // var propertyDisplayName = property.displayName;
             _.each($scope.allVariationPropsMap[property.displayName], function (x) {
                 x.selected = x != property ? false : !x.selected;
             });
 
+            _.each(_.keys($scope.allVariationPropsMap), function (key) {
+                _.each($scope.allVariationPropsMap[key], function (prop) {
+
+                     if($scope.allVariationPropsMap[key].length == 1) {
+                         $scope.allVariationPropsMap[key][0].selected = true;
+                    }
+                    else {
+                        var available = isPropertyAvailable(prop, key);
+                        var originalProperty = _.find($scope.allVariationPropsMap[key], function (y) { return y.value === prop.value; });
+                        originalProperty.available = available;
```

Comment:
Can we use this form? This is more readable:
originalProperty.available = isPropertyAvailable(prop, key);